python/src/lib.rs

```diff
@@ -85,6 +85,14 @@ impl Expr {
     fn to_sql(&self) -> Result<SqlQuery> {
         self.0.to_sql().map(SqlQuery::from).map_err(Error::from)
     }
+
+    fn __add__(&self, rhs: &Expr) -> Result<Expr> {
+        Ok(Expr(self.0.clone() + rhs.0.clone()))
+    }
+
+    fn __eq__(&self, rhs: &Expr) -> bool {
+        self.0 == rhs.0
+    }
```

Comment:
You also have the option of putting
```
#[derive(Eq)]
```
on the `pyclass`:
https://pyo3.rs/v0.23.4/class.html?highlight=partialeq#customizing-the-class